src/__tests__/geo.test.js

```diff
@@ -380,3 +575,23 @@ describe('Provides region name for a given region code', () => {
     expect(findRegionFromRegionCode(33).name).toEqual('France');
   });
 });
+
+describe('Extracts all useful phone info from a long string of text', () => {
+  it('Returns all phone numbers and geo data', () => {
+    const numbers = findNumbersInString(longStringWithNumbers);
+    const goodNumbers = numbers.map((item) => {
+      if (item.areaCode) {
+        const geo = findTimeFromAreaCode(
+          item.areaCode,
+          new Date('2024-07-20T08:00:00'),
+        );
+        return { ...item, ...geo };
+      } else {
+        const geo = findRegionFromRegionCode(item.regionCode);
+        return { ...item, ...geo };
```

Comment:
After taking another look at this, I think we want to avoid leveraging so much logic that seems testable in itself as part of a test.  Is there any way you can avoid this or maybe make a test utility that can also be tested itself?